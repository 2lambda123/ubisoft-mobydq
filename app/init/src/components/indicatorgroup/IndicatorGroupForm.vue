<template>
  <div>
    <h1 class="mt-5">Edit Indicator Group</h1>

    <form>
      <div class="form-row">
        <div class="col-md-4">
          <!-- Data Source Form -->
          <div class="form-group required">
            <label for="indicatorGroupName" class="col-form-label">
              Name:
            </label>
            <input
              class="form-control col-sm"
              id="indicatorGroupName"
              type="text"
              required="required"
              placeholder="Type indicator group name"
              v-model="indicatorGroup.name"
            />
          </div>

          <!-- Meta-Data -->
          <div>
            <indicator-group-meta-data
              v-if="indicatorGroup.id"
              v-bind:id="indicatorGroup.id"
              v-bind:createdDate="indicatorGroup.createdDate"
              v-bind:createdBy="indicatorGroup.userByCreatedById.email"
              v-bind:updatedDate="indicatorGroup.updatedDate"
              v-bind:updatedBy="indicatorGroup.userByUpdatedById.email"
            ></indicator-group-meta-data>
          </div>

          <!-- Button Menu -->
          <div class="mt-3">
            <indicator-group-button-save v-bind:indicatorGroup="indicatorGroup"> </indicator-group-button-save>
            <indicator-group-button-close> </indicator-group-button-close>
            <indicator-group-button-delete v-if="indicatorGroup.id" v-bind:indicatorGroupId="indicatorGroup.id"> </indicator-group-button-delete>
          </div>
        </div>
      </div>
    </form>
  </div>
</template>

<script>
import IndicatorGroupButtonSave from "./IndicatorGroupButtonSave.vue";
import IndicatorGroupButtonClose from "./IndicatorGroupButtonClose.vue";
import IndicatorGroupButtonDelete from "./IndicatorGroupButtonDelete.vue";
import MetaDataCard from "../utils/MetaDataCard.vue";
import Mixins from "../utils/Mixins.vue";

export default {
  mixins: [Mixins],
  components: {
    "indicator-group-meta-data": MetaDataCard,
    "indicator-group-button-save": IndicatorGroupButtonSave,
    "indicator-group-button-close": IndicatorGroupButtonClose,
    "indicator-group-button-delete": IndicatorGroupButtonDelete
  },
  data: function() {
    return {
      indicatorGroup: {}
    };
  },
  computed: {
    indicatorGroupId() {
      return this.$route.params.indicatorGroupId;
    }
  },
  created: function() {
    // If indicatorGroupId != new then get data for existing indicator group
    if (this.indicatorGroupId != "new") {
      let payload = {
        query: this.$store.state.queryGetIndicatorGroup,
        variables: {
          id: parseInt(this.indicatorGroupId)
        }
      };
      let headers = {};
      if (this.$session.exists()) {
        headers = { Authorization: "Bearer " + this.$session.get("jwt") };
      }
      this.$http.post(this.$store.state.graphqlUrl, payload, { headers }).then(
        function(response) {
          if (response.data.errors) {
            this.displayError(response);
          } else {
            this.indicatorGroup = response.data.data.indicatorGroupById;
          }
        },
        // Error callback
        function(response) {
          this.displayError(response);
        }
      );
    }
  }
};
</script>
