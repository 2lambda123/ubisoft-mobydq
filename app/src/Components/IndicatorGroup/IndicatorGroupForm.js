import React from 'react';
import { withFormik } from 'formik';
import * as Yup from 'yup';
import TextInput from './../FormInput/TextInput';
import Button from '@material-ui/core/Button';
import SaveIcon from '@material-ui/icons/Save';

const IndicatorGroupFormFields = (props) => {
  const {
    values,
    touched,
    errors,
    handleChange,
    handleBlur,
    handleSubmit,
    isSubmitting
  } = props;
  return <form onSubmit={handleSubmit} style={{ 'marginLeft': '60px' }}>
    <div>
      <TextInput
        id="name"
        label="Name"
        helperText=""
        placeholder=""
        touched={touched.name}
        error={touched.name && errors.name}
        value={values.name}
        onChange={handleChange}
        onBlur={handleBlur}
      />
    </div>
    <div>
      <TextInput
        id="createdDate"
        label="Created Date"
        helperText=""
        placeholder=""
        touched={touched.createdDate}
        error={touched.createdDate && errors.createdDate}
        value={values.createdDate}
        onChange={handleChange}
        onBlur={handleBlur}
        disabled={true}
      />
    </div>
    <div>
      <TextInput
        id="createdBy"
        label="Created By"
        helperText=""
        placeholder=""
        touched={touched.createdBy}
        error={touched.createdBy && errors.createdBy}
        value={values.createdBy}
        onChange={handleChange}
        onBlur={handleBlur}
        disabled={true}
      />
    </div>
    <div>
      <TextInput
        id="updatedDate"
        label="Updated Date"
        helperText=""
        placeholder=""
        touched={touched.updatedDate}
        error={touched.updatedDate && errors.updatedDate}
        value={values.updatedDate}
        onChange={handleChange}
        onBlur={handleBlur}
        disabled={true}
      />
    </div>
    <div>
      <TextInput
        id="updatedBy"
        label="Updated By"
        helperText=""
        placeholder=""
        touched={touched.updatedBy}
        error={touched.updatedBy && errors.updatedBy}
        value={values.updatedBy}
        onChange={handleChange}
        onBlur={handleBlur}
        disabled={true}
      />
    </div>
    <div>
      <Button type="submit" disabled={isSubmitting} variant="contained" color={'secondary'}>
        <SaveIcon />
        Save
      </Button>
    </div>
  </form>;
};

const formikEnhancer = withFormik({
  'validationSchema': Yup.object().shape({
    'name': Yup.string()
      .required('Name cannot be blank')
  }),
  'mapPropsToValues': (props) => props.initialFieldValues === null
    ? {
        'name': ''
      }
    : {
        'name': props.initialFieldValues.name,
        'createdDate': props.initialFieldValues.createdDate,
        'createdBy': props.initialFieldValues.userByCreatedById.email,
        'updatedDate': props.initialFieldValues.updatedDate,
        'updatedBy': props.initialFieldValues.userByUpdatedById.email
      },
  'handleSubmit': (payload, { props, setSubmitting }) => {
    setSubmitting(false);
    let variables;
    if (props.initialFieldValues === null) {
      variables = { 'indicatorGroup': payload };
    } else {
      variables = { 'indicatorGroupPatch': payload, 'id': props.initialFieldValues.id };
    }
    props.mutate({
      variables
    });
  },
  'displayName': 'IndicatorGroupForm'
});


const EnhancedIndicatorGroupForm = formikEnhancer(IndicatorGroupFormFields);

export default EnhancedIndicatorGroupForm;
