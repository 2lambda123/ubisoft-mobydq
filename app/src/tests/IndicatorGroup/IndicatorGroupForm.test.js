import React from 'react';
import { shallowWrap } from './../../setupTests';

import IndicatorGroupForm from './../../Components/IndicatorGroup/IndicatorGroupForm';

describe('IndicatorGroupForm component unit test', () => {
  const data = {};

  const wrapper = shallowWrap(<IndicatorGroupForm data={data} initialFieldValues={null} handleBlur={jest.fn()} handleChange={jest.fn()} handleSubmit={jest.fn()} isSubmitting={false}/>);
  it('matches snapshot', () => {
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
