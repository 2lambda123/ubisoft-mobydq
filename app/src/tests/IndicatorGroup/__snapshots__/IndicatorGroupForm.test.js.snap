// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndicatorGroupForm component unit test matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Formik
    data={Object {}}
    displayName="IndicatorGroupForm"
    enableReinitialize={false}
    handleBlur={[MockFunction]}
    handleChange={[MockFunction]}
    handleSubmit={[Function]}
    initialFieldValues={null}
    initialValues={
      Object {
        "name": "",
      }
    }
    isInitialValid={false}
    isSubmitting={false}
    onSubmit={[Function]}
    render={[Function]}
    validateOnBlur={true}
    validateOnChange={true}
    validationSchema={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <IndicatorGroupFormFields
        data={Object {}}
        dirty={false}
        errors={Object {}}
        handleBlur={[Function]}
        handleChange={[Function]}
        handleReset={[Function]}
        handleSubmit={[Function]}
        initialFieldValues={null}
        initialValues={
          Object {
            "name": "",
          }
        }
        isSubmitting={false}
        isValid={false}
        isValidating={false}
        registerField={[Function]}
        resetForm={[Function]}
        setError={[Function]}
        setErrors={[Function]}
        setFieldError={[Function]}
        setFieldTouched={[Function]}
        setFieldValue={[Function]}
        setFormikState={[Function]}
        setStatus={[Function]}
        setSubmitting={[Function]}
        setTouched={[Function]}
        setValues={[Function]}
        submitCount={0}
        submitForm={[Function]}
        touched={Object {}}
        unregisterField={[Function]}
        validateField={[Function]}
        validateForm={[Function]}
        validateOnBlur={true}
        validateOnChange={true}
        values={
          Object {
            "name": "",
          }
        }
      />,
      "value": Object {
        "dirty": false,
        "errors": Object {},
        "handleBlur": [Function],
        "handleChange": [Function],
        "handleReset": [Function],
        "handleSubmit": [Function],
        "initialValues": Object {
          "name": "",
        },
        "isSubmitting": false,
        "isValid": false,
        "isValidating": false,
        "registerField": [Function],
        "resetForm": [Function],
        "setError": [Function],
        "setErrors": [Function],
        "setFieldError": [Function],
        "setFieldTouched": [Function],
        "setFieldValue": [Function],
        "setFormikState": [Function],
        "setStatus": [Function],
        "setSubmitting": [Function],
        "setTouched": [Function],
        "setValues": [Function],
        "status": undefined,
        "submitCount": 0,
        "submitForm": [Function],
        "touched": Object {},
        "unregisterField": [Function],
        "validate": undefined,
        "validateField": [Function],
        "validateForm": [Function],
        "validateOnBlur": true,
        "validateOnChange": true,
        "validationSchema": [Function],
        "values": Object {
          "name": "",
        },
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "data": Object {},
        "dirty": false,
        "errors": Object {},
        "handleBlur": [Function],
        "handleChange": [Function],
        "handleReset": [Function],
        "handleSubmit": [Function],
        "initialFieldValues": null,
        "initialValues": Object {
          "name": "",
        },
        "isSubmitting": false,
        "isValid": false,
        "isValidating": false,
        "registerField": [Function],
        "resetForm": [Function],
        "setError": [Function],
        "setErrors": [Function],
        "setFieldError": [Function],
        "setFieldTouched": [Function],
        "setFieldValue": [Function],
        "setFormikState": [Function],
        "setStatus": [Function],
        "setSubmitting": [Function],
        "setTouched": [Function],
        "setValues": [Function],
        "status": undefined,
        "submitCount": 0,
        "submitForm": [Function],
        "touched": Object {},
        "unregisterField": [Function],
        "validateField": [Function],
        "validateForm": [Function],
        "validateOnBlur": true,
        "validateOnChange": true,
        "values": Object {
          "name": "",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": Object {
      "$$typeof": Symbol(react.provider),
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": Object {
          "$$typeof": Symbol(react.context),
          "_calculateChangedBits": null,
          "_context": [Circular],
        },
        "Provider": [Circular],
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": Object {},
        "_currentValue2": Object {},
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <IndicatorGroupFormFields
          data={Object {}}
          dirty={false}
          errors={Object {}}
          handleBlur={[Function]}
          handleChange={[Function]}
          handleReset={[Function]}
          handleSubmit={[Function]}
          initialFieldValues={null}
          initialValues={
            Object {
              "name": "",
            }
          }
          isSubmitting={false}
          isValid={false}
          isValidating={false}
          registerField={[Function]}
          resetForm={[Function]}
          setError={[Function]}
          setErrors={[Function]}
          setFieldError={[Function]}
          setFieldTouched={[Function]}
          setFieldValue={[Function]}
          setFormikState={[Function]}
          setStatus={[Function]}
          setSubmitting={[Function]}
          setTouched={[Function]}
          setValues={[Function]}
          submitCount={0}
          submitForm={[Function]}
          touched={Object {}}
          unregisterField={[Function]}
          validateField={[Function]}
          validateForm={[Function]}
          validateOnBlur={true}
          validateOnChange={true}
          values={
            Object {
              "name": "",
            }
          }
        />,
        "value": Object {
          "dirty": false,
          "errors": Object {},
          "handleBlur": [Function],
          "handleChange": [Function],
          "handleReset": [Function],
          "handleSubmit": [Function],
          "initialValues": Object {
            "name": "",
          },
          "isSubmitting": false,
          "isValid": false,
          "isValidating": false,
          "registerField": [Function],
          "resetForm": [Function],
          "setError": [Function],
          "setErrors": [Function],
          "setFieldError": [Function],
          "setFieldTouched": [Function],
          "setFieldValue": [Function],
          "setFormikState": [Function],
          "setStatus": [Function],
          "setSubmitting": [Function],
          "setTouched": [Function],
          "setValues": [Function],
          "status": undefined,
          "submitCount": 0,
          "submitForm": [Function],
          "touched": Object {},
          "unregisterField": [Function],
          "validate": undefined,
          "validateField": [Function],
          "validateForm": [Function],
          "validateOnBlur": true,
          "validateOnChange": true,
          "validationSchema": [Function],
          "values": Object {
            "name": "",
          },
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "data": Object {},
          "dirty": false,
          "errors": Object {},
          "handleBlur": [Function],
          "handleChange": [Function],
          "handleReset": [Function],
          "handleSubmit": [Function],
          "initialFieldValues": null,
          "initialValues": Object {
            "name": "",
          },
          "isSubmitting": false,
          "isValid": false,
          "isValidating": false,
          "registerField": [Function],
          "resetForm": [Function],
          "setError": [Function],
          "setErrors": [Function],
          "setFieldError": [Function],
          "setFieldTouched": [Function],
          "setFieldValue": [Function],
          "setFormikState": [Function],
          "setStatus": [Function],
          "setSubmitting": [Function],
          "setTouched": [Function],
          "setValues": [Function],
          "status": undefined,
          "submitCount": 0,
          "submitForm": [Function],
          "touched": Object {},
          "unregisterField": [Function],
          "validateField": [Function],
          "validateForm": [Function],
          "validateOnBlur": true,
          "validateOnChange": true,
          "values": Object {
            "name": "",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": Object {
        "$$typeof": Symbol(react.provider),
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": Object {
            "$$typeof": Symbol(react.context),
            "_calculateChangedBits": null,
            "_context": [Circular],
          },
          "Provider": [Circular],
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": Object {},
          "_currentValue2": Object {},
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "childContextTypes": Object {
      "router": [Function],
    },
    "context": Object {
      "router": Object {
        "history": Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        },
        "route": Object {
          "location": Object {},
          "match": Object {},
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
